src/components/flyout/OverlayHeader.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+
+import classNames from 'classnames';
+import CloseButton from '../close-button/CloseButton';
+// @ts-ignore flow
+import { FlyoutContext } from './Flyout';
+
+import './OverlayHeader.scss';
+
+export interface OverlayHeaderProps {
+    /** Components to render in the header */
+    children?: React.ReactNode;
+    /** Set className to the overlay header */
+    className?: string;
+}
+
+const OverlayHeader = ({ children, className }: OverlayHeaderProps) => {
```

Comment:
Do we need tests for this component?

Reply:
Yes, I'm going to commit that separately. I had to change a lot of things, so I wanted to focus on tests after

Reply:
When you compose that test, let's make sure it's RTL instead of enzyme (c.f. comment above). I can help with that :)